go/enclave/components/batch_producer.go

```diff
@@ -0,0 +1,202 @@
+	if err != nil {
+		bp.logger.Error("Extracting messages L2->L1 failed", err, log.CmpKey, log.CrossChainCmp)
+		return fmt.Errorf("could not extract cross chain messages. Cause: %w", err)
+	}
+
+	batch.Header.CrossChainMessages = crossChainMessages
+
+	bp.logger.Trace(fmt.Sprintf("Added %d cross chain messages to batch.",
+		len(batch.Header.CrossChainMessages)), log.CmpKey, log.CrossChainCmp)
+
+	batch.Header.LatestInboundCrossChainHash = block.Hash()
+	batch.Header.LatestInboundCrossChainHeight = block.Number()
+
+	return nil
+}
+
+func (bp *batchProducer) populateHeader(batch *core.Batch, receipts types.Receipts) {
+	if len(receipts) == 0 {
+		batch.Header.ReceiptHash = types.EmptyRootHash
+	} else {
+		batch.Header.ReceiptHash = types.DeriveSha(receipts, trie.NewStackTrie(nil))
+		batch.Header.Bloom = types.CreateBloom(receipts)
+	}
+
+	if len(batch.Transactions) == 0 {
+		batch.Header.TxHash = types.EmptyRootHash
+	} else {
+		batch.Header.TxHash = types.DeriveSha(types.Transactions(batch.Transactions), trie.NewStackTrie(nil))
+	}
+}
+
+func (bp *batchProducer) ComputeBatch(context *BatchContext) (*ComputedBatch, error) {
+	// These variables will be used to create the new batch
+
+	parent, err := bp.storage.FetchBatch(context.ParentPtr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve parent batch. Cause: %w", err)
+	}
+
+	block, err := bp.storage.FetchBlock(context.BlockPtr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve block for batch. Cause: %w", err)
+	}
+
+	parentBlock := block
+	if parent.Header.L1Proof != block.Hash() {
+		var err error
+		parentBlock, err = bp.storage.FetchBlock(parent.Header.L1Proof)
+		if err != nil {
+			bp.logger.Crit(fmt.Sprintf("Could not retrieve a proof for batch %s", parent.Hash()), log.ErrKey, err)
+		}
+	}
+
+	// Create a new batch based on the fromBlock of inclusion of the previous, including all new transactions
+	batch := core.DeterministicEmptyBatch(context.Creator, parent.Header, block, context.Randomness, context.AtTime)
+
+	stateDB, err := bp.storage.CreateStateDB(batch.Header.ParentHash)
+	if err != nil {
+		return nil, fmt.Errorf("could not create stateDB. Cause: %w", err)
+	}
+
+	messages := bp.crossChainProcessors.Local.RetrieveInboundMessages(parentBlock, block, stateDB)
+	crossChainTransactions := bp.crossChainProcessors.Local.CreateSyntheticTransactions(messages, stateDB)
+
+	successfulTxs, txReceipts, err := bp.processTransactions(batch, 0, context.Transactions, stateDB, context.ChainConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ccSuccessfulTxs, ccReceipts, err := bp.processTransactions(batch, len(successfulTxs), crossChainTransactions, stateDB, context.ChainConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := bp.verifyInboundCrossChainTransactions(crossChainTransactions, ccSuccessfulTxs, ccReceipts); err != nil {
+		return nil, fmt.Errorf("batch computation failed due to cross chain messages. Cause: %w", err)
```

Comment:
This shouldn't be possible, we generate the synthetic transactions out of the rlp encoded events. It's just a data transfer so the data inside the message makes no difference whatsoever since it does not trigger any arbitary executions. Perhaps if the data is insanely big (but someone has to pay for this on l1) and l1 allows for a transaction with more size than we do. I need to think about it more, but in general there shouldn't be anything DoS worthy here.